src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -426,6 +426,12 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
 
             foreach (var endpoint in workloadInfo.ReachableEndpoints)
             {
+                if (endpoint.Ports.Length == 0)

```

Comment:
@ababilone Thanks for this PR, can we add a check ```if (endpoint.ports == null || endpoint.Ports.Length == 0)``` just in case if the ports element is null?